src/xgui/debuga/debuger.cpp

```diff
@@ -1310,11 +1310,12 @@ void DebugWin::chLayout() {
 	}
 }
 
-void DebugWin::regClick(QMouseEvent* ev) {
-	xLabel* lab = (xLabel*)sender();
+void DebugWin::regClick(QMouseEvent* ev) {	  
+	xLabel* lab = qobject_cast<xLabel*>(sender());
+	if ( lab == NULL ) return;	
 	int id = lab->id;
-	if (id < 0) return;
-	if (id > 15) return;
+	// Checking id is in range [0..15]
+	if ( !( (id-15)*(id-0) <= 0) ) return;
```

Comment:
Sorry for the unsolicited comment, but I think the previous variant was more readable.
What's wrong with `if (id < 0 || id > 15) return;` ?

Reply:
Okay, your version reads better.
However, checking the range at this place indicates an architectural problem.
At first, https://github.com/samstyle/Xpeccy/blob/15c7cadbb271e99a0123e7d1b72726a8e02d35fb/src/xgui/debuga/debuger.cpp#L279
defined as array with 16 item's and item with index=16 defined as NULL
https://github.com/samstyle/Xpeccy/blob/15c7cadbb271e99a0123e7d1b72726a8e02d35fb/src/xgui/debuga/debuger.cpp#L283
At second, inside the loop https://github.com/samstyle/Xpeccy/blob/15c7cadbb271e99a0123e7d1b72726a8e02d35fb/src/xgui/debuga/debuger.cpp#L292
connect https://github.com/samstyle/Xpeccy/blob/15c7cadbb271e99a0123e7d1b72726a8e02d35fb/src/xgui/debuga/debuger.cpp#L297
at third, call SLOT https://github.com/samstyle/Xpeccy/blob/15c7cadbb271e99a0123e7d1b72726a8e02d35fb/src/xgui/debuga/debuger.cpp#L1313
if `xLabel* lab = (xLabel*)sender();` lab == VALID
id always in the range [0..15]

Reply:
It's just unnecessary checks. This emulator is in 'always beta-version' state. Some sh*t  is possible.